src/App.jsx

```diff
@@ -19,6 +19,7 @@ function App() {
         <NavBar />
         <Routes>
           <Route path="/" element={<ArticleList />} />
+          <Route path="/articles/:topic" element={<ArticleList />} />
           <Route path="/articles/:article_id" element={<ArticlePage/>}/>
```

Comment:
You've implemented topics correctly and it works, but now when I click on a single article it renders the same page. It's because you have two matching routes here. React won't tell the difference between :topic or :article_id because they are just parameters and treats the two endpoints as the same, so I currently get the topics page loading when I wanted to go to the single article page. 

If you change one of these slightly and match that wherever your links are, it should be fine :)  /articles/singleArticle/:article_id for example - it would be a different route to topics then.